samples/lightweight-component/lightweight_component.ipynb

```diff
@@ -0,0 +1,359 @@
+    "#### Convert the function to a pipeline operation\n",
+    "\n",
+    "You can specify an alternative base container image (the image needs to have Python 3.5+ installed)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "divmod_op = comp.func_to_container_op(my_divmod, base_image='tensorflow/tensorflow:1.11.0-py3')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define the pipeline\n",
+    "Pipeline function has to be decorated with the `@dsl.pipeline` decorator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp.dsl as dsl\n",
+    "@dsl.pipeline(\n",
+    "   name='Calculation pipeline',\n",
+    "   description='A toy pipeline that performs arithmetic calculations.'\n",
+    ")\n",
+    "# Currently kfp-tekton doesn't support pass parameter to the pipelinerun yet, so we hard code the number here\n",
+    "def calc_pipeline(\n",
+    "   name=pipeline_name,\n",
+    "   namespace=user_namespace, \n",
+    "   a='7',\n",
+    "   b='8',\n",
+    "   c='17',\n",
+    "):\n",
+    "    #Passing pipeline parameter and a constant value as operation arguments\n",
+    "    add_task = add_op(a, 4) #Returns a dsl.ContainerOp class instance. \n",
+    "    \n",
+    "    #Passing a task output reference as operation arguments\n",
+    "    #For an operation with a single return value, the output reference can be accessed using `task.output` or `task.outputs['output_name']` syntax\n",
+    "    divmod_task = divmod_op(add_task.output, b)\n",
+    "\n",
+    "    #For an operation with a multiple return values, the output references can be accessed using `task.outputs['output_name']` syntax\n",
+    "    result_task = add_op(divmod_task.outputs['quotient'], c)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# !kubectl delete pipelinerun calculation-pipeline-run -n $user_namespace"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Compile the pipeline into Tekton yaml using kfp-tekton SDK"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from kfp_tekton.compiler import TektonCompiler\n",
+    "TektonCompiler().compile(calc_pipeline, 'calc.yaml', generate_pipelinerun=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Assign permission to Tekton's default service account and run this pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl create clusterrolebinding $user_namespace-edit --clusterrole kubeflow-edit --serviceaccount=$user_namespace:default\n",
```

Comment:
We probably don't need this because we are not using any k8s api.

Reply:
sure